### PR TITLE
Refactor: Check for server status in stateful node test

### PR DESCRIPTION
**Pull request by lwojtow**

### CODE REVIEW

File: src/tools/rpc.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    Client::new()
+        .post(JSON_RPC_ADDRESS)
+        .header(CONTENT_TYPE, "application/json")
+        .header(ACCEPT, "application/json")
+        .json(request)
+}
+
+fn build_server_info_request() -> RpcRequest {
+    RpcRequest {
+        id: "1".to_string(),
```
Comment (niklaslong) [inline src/tools/rpc.rs:40]:
```suggestion
        id: String::from("1"),
```

File: src/tools/rpc.rs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        method: "server_info".to_string(),
```
Comment (niklaslong) [inline src/tools/rpc.rs:41]:
```suggestion
        method: String::from("server_info"),
```

File: src/tools/rpc.rs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        api_version: 1,
```
Comment (niklaslong) [inline src/tools/rpc.rs:42]: We may want this to be a constant?

File: src/tools/rpc.rs (hunk 1 above)
Comment (lwojtow) [reply src/tools/rpc.rs]: Done

File: src/tools/rpc.rs (hunk 2 above)
Comment (lwojtow) [reply src/tools/rpc.rs]: Done

File: src/tools/rpc.rs (hunk 3 above)
Comment (lwojtow) [reply src/tools/rpc.rs]: Done

**Review by niklaslong — APPROVED** [review]

**Pull request merged**